File: src/frontend/src/lib/types/bullet.ts
import type { Position } from './position';
import type { Tank } from './tank';

export type Bullet = {
	Id: string;
	Shooter: Tank;
	CurrentPosition: Position;
	Status: BulletStatus;
};

export enum BulletStatus {
	ShotStart,
	InFlight,
	Hit,
	SuperHit
}

File: src/backend/BotBattle.Core/Bullet.cs
using BotBattle.Core.Enums;

namespace BotBattle.Core;

public class Bullet
{
    public required string Id { get; init; }
    public required Tank Shooter { get; init; }
    public required Position CurrentPosition { get; set; }
    public required int ShootingRange { get; set; }
    public BulletStatus Status { get; set; } = BulletStatus.ShotStart;
    public BulletType Type { get; init; }
}